### PR TITLE
[PECO-1926] Create a non pyarrow flow to handle small results for the column set

### PR DESCRIPTION
**Pull request by jprakash-db**
[PECO-1926]

## Description

To handle the flow of result when pyarrow is not present. This setup is particularly used for users who want to handle small quantities of data 

## Task Completed

- [x] Introduced the new non arrow flow for data to be handled by the users not having pyarrow
- [x] ColumnQueue is the new ResultSetQueue that I have added to handle the column set data 
- [x] Entire pipeline of handling the column set data to finally converting it to the TRow type for the user has been implemented
- [x] Added tests for the ColumnQueue to ensure that it is working as intended

[PECO-1926]: https://databricks.atlassian.net/browse/PECO-1926?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/databricks/sql/utils.py [hunk 1]
```diff
@@ -96,11 +102,33 @@ def build_queue(
...
+        length = min(num_rows, self.n_valid_rows - self.cur_row_index)
+        # Slicing using the default python slice
+        next_data = [
+            column[self.cur_row_index : self.cur_row_index + length]
+            for column in self.columnar_table
+        ]
+        self.cur_row_index += length
+        return next_data
+
+    def remaining_rows(self):
```
Comment (jackyhu-db) [inline src/databricks/sql/utils.py:122]: Is this standard naming? The name seems this is read only operation

File: src/databricks/sql/client.py [hunk 2]
```diff
@@ -1210,7 +1227,42 @@ def fetchmany_arrow(self, size: int) -> pyarrow.Table:
...
+    def fetchmany_columnar(self, size: int):
+        """
+        Fetch the next set of rows of a query result, returning a Columnar Table.
+        An empty sequence is returned when no more rows are available.
+        """
+        if size < 0:
+            raise ValueError("size argument for fetchmany is %s but must be >= 0", size)
+
+        results = self.results.next_n_rows(size)
+        n_remaining_rows = size - len(results[0])
```
Comment (jackyhu-db) [inline src/databricks/sql/client.py:1249]: what if there is no column in the result?

File: src/databricks/sql/utils.py [hunk 3]
```diff
@@ -560,6 +588,29 @@ def convert_decimals_in_arrow_table(table, description) -> pyarrow.Table:
     return table
 
 
+def convert_to_assigned_datatypes_in_column_table(column_table, description):
+    for i, col in enumerate(column_table):
```
Comment (jackyhu-db) [inline src/databricks/sql/utils.py:592]: I would recommend not change `column_table`, instead, return a table with new assigned data types, input argument should be immutable, if the function crashes in the middle, the state of column_table becomes inconsisent.

File: src/databricks/sql/utils.py [hunk 4]
```diff
@@ -595,6 +653,26 @@ def _convert_column_to_arrow_array(t_col):
     raise OperationalError("Empty TColumn instance {}".format(t_col))
 
 
+def _covert_column_to_list(t_col):
+    supported_field_types = (
```
Comment (jackyhu-db) [inline src/databricks/sql/utils.py:657]: can we define it as the static or constant variable if this is constant.

File: src/databricks/sql/utils.py (hunk 1 above)
Comment (jprakash-db) [reply src/databricks/sql/utils.py:122]: The names are defined in the already existing ResultSetQueue from which I inherited

File: src/databricks/sql/utils.py (hunk 4 above)
Comment (jprakash-db) [reply src/databricks/sql/utils.py:657]: The function is not part of a class, so I don't think we can make it static. In Python I believe we can't make anything constant, so I have replaced the names with ALL CAPS to indicate that this is a constant

File: src/databricks/sql/utils.py (hunk 3 above)
Comment (jprakash-db) [reply src/databricks/sql/utils.py:592]: Fixed it

**Review by jackyhu-db — APPROVED** [review]

File: src/databricks/sqlalchemy/pytest.ini [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply src/databricks/sqlalchemy/pytest.ini]: Removed it

File: src/databricks/sql/thrift_backend.py [hunk 6]
```diff
@@ -621,6 +624,12 @@ def _get_metadata_resp(self, op_handle):
 
     @staticmethod
     def _hive_schema_to_arrow_schema(t_table_schema):
+
+        if pyarrow is None:
+            raise ImportError(
+                "pyarrow is required to convert Hive schema to Arrow schema"
+            )
+
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply src/databricks/sql/thrift_backend.py]: fixed

**Review by kravets-levko — APPROVED** [review]

**Pull request merged**